force-app/main/default/classes/SOQL.cls

```diff
@@ -334,28 +394,10 @@ public with sharing class SOQL {
      * Example: getOpportunityDetailsForClosedWon()
      * @return A list of filtered Opportunity
      */
-    public static List<Opportunity> getOpportunityDetailsForClosedWon() {        
-        return null; // Replace null with the variable you used to store the result
-    }
-
-    /**
-     * Question 25
-     * Sets the 'Primary_Contact__c' field for all Closed Won Opportunities to first Contact on the Account.
-     * Example: setPrimaryContact() should set the 'Primary_Contact__c' field for every Opportunity in the org
-     */
-    public static void setPrimaryContact() {
-        //Use the getOpportunityDetailsForClosedWon() method to get the Opportunities
-
-        // Create a set of Account Ids from the Opportunities
-
-        // Query the Contacts
-
-        // Loop through each Contact and add the Account Id and Contact to the map
-        
-        // Loop through each Opportunity
-
-                // Update the Opportunity primary contact from accountToContactMap
-
-        // Update the Opportunities
-    }
+    public static List<Opportunity> getOpportunityDetailsForClosedWon() {    
+        List<Opportunity> closedWonOpps = [SELECT Name, Id, Account.Name, Owner.Name FROM Opportunity WHERE StageName = 'Closed Won'];    
+        return closedWonOpps; // Replace null with the variable you used to store the result
+ 
+    }  
+    
```

Comment:
Looks like you're missing Question 25. This was a tricky one. If you want to arrange a 1-on-1 with me (or someone else) it would be worth your time. Also, the wording is confusing. You can think of the "first contact" as being the first one to appear alphabetically.